### ISSUE TITLE
rspack v0.3.4 cannot be started in mono-repo when using workspaces in package.json

### ISSUE THREAD

**Issue by erikvullings** [web-infra-dev/rspack#4203]
### System Info

  System:
    OS: Linux 5.15 Ubuntu 22.04.3 LTS 22.04.3 LTS (Jammy Jellyfish)
    CPU: (7) x64 Intel(R) Core(TM) i7-10850H CPU @ 2.70GHz
    Memory: 11.45 GB / 19.54 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 16.20.2 - /usr/bin/node
    Yarn: 1.22.19 - /usr/bin/yarn
    npm: 8.19.4 - /usr/bin/npm
    pnpm: 8.7.5 - ~/dev/cat/node_modules/.bin/pnpm
  Browsers:
    Chrome: 107.0.5304.87

### Details

I've used rspack in a number of projects with impressive results, and I really like it. However, when installing it with `pnpm` in a simple [mono-repo](https://github.com/TNO/cat) to build my GUI in `packages/gui`, a simple command like `npx rspack --version` failed:

```bash
npm ERR! could not determine executable to run
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/erik/.npm/_logs/2023-09-16T15_36_22_686Z-debug-0.log
```

The root cause, I assume, is that my root `package.json` contains a `"workspaces": ["packages/*"]`. Without it, it runs just fine.

### Reproduce Steps

1. mkdir -p tmp/packages/gui
2. cd tmp
3. npm init -y  
4. edit `package.json` and add `"workspaces": [ "packages/*" ],`
5. cd packages/gui
6. npm init -y
7. pnpm i -D @rspack/cli
8. npx rspack --version

Root `package.json` with the offending `workspaces`.

```json
{
  "name": "tmp",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "workspaces": [
    "packages/*"
  ]
}
```

**Comment by hardfist** [comment]
@erikvullings  this seems not a rspack bug but package manage bug, and I following your steps here and everything works fine
![image](https://github.com/web-infra-dev/rspack/assets/8898718/589d3dfa-f3e3-461a-9865-09ad47ea479c)
maybe you can help provide a repro demo so we can investigate futher

**Comment by erikvullings** [comment]
Thanks for the quick reply, and indeed, you are right: it is a `pnpm` issue. I've created a test repo `git clone https://github.com/erikvullings/rspack-test.git`, and when using `pnpm i` to install all dependencies, it fails to run `npx rspack --version`. Using `npm i` works, however, as well as `bun i`. 

Interestingly, I tried to run the same test on Ubuntu (not in WSL2), using the same v8.7.5 of `pnpm`, and it warned me that the `workspaces` property is not supported in `pnpm` (which it did not in WSL2). So I will close the issue here and raise it there...
